src/Sparkfabrik/Command/Redmine/RedmineGitCommitCommand.php

```diff
@@ -0,0 +1,167 @@
+
+class RedmineGitCommitCommand extends RedmineCommand
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('redmine:git:commit')
+            ->setDescription(<<<EOF
+Generate git commit message starting from issue subject
+EOF
+            );
+        $this->addArgument(
+            'issue',
+            InputArgument::REQUIRED,
+            'Issue id'
+        );
+        $this->addOption(
+            'execute',
+            'e',
+            InputOption::VALUE_NONE,
+            'Print the Commit message and execute the commit too.'
+        );
+        $this->addOption(
+            'no-verify',
+            false,
+            InputOption::VALUE_NONE,
+            'Escapes the CS check.'
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $base_msg = $this->getService()->getConfig()['commit_pattern'];
+        $client = $this->getService()->getClient();
+        $issue = $input->getArgument('issue');
+        $execute = $input->getOption('execute');
+        $skipcs = $input->getOption('no-verify');
+        $res = $client->api('issue')->show($issue);
+
+        // Handle errors.
+        if (isset($res['errors'])) {
+            $errors = implode("\n", $res['errors']);
+            throw new \Exception($errors);
+        }
+        if ($res === 1) {
+            return $output->writeln('<info>No issues found.</info>');
+        }
+
+        // Use Custom field to determine the story name.
+        $story = null;
+        $story_name = $res['issue']['subject'];
+        foreach ($res['issue']['custom_fields'] as $field) {
+            if ($field['name'] === 'Jira Story Code') {
+                $story = $field['value'];
+                $story_name = str_replace($field['value'], '', $story_name);
+                $story_name = trim($story_name);
+            }
+        }
+
+        // Stop execution if the JIRA Code field is empty.
+        if (empty($story)) {
+            return $output->writeln('<info>The Issue is not consistent, please compiler the Jira Story Code.</info>');
+        }
+
+        // Clean up story name.
+        if (mb_detect_encoding($story_name) === 'UTF-8') {
+            $story_name_converted = @iconv('UTF-8', 'ASCII//TRANSLIT', $story_name);
+            if ($story_name_converted) {
+                $story_name = $story_name_converted;
+            } else {
+                $story_name = iconv('UTF-8', 'ASCII//IGNORE', $story_name);
+            }
+        }
+        $story_name = preg_replace('/[a-z0-9 ]{0,}[- ]{0,}/', '', $story_name, 1);
+
+        // Output to the user the current status of informations.
+        $output->writeln('<info>Issue Number</info>: ' . $issue);
+        $output->writeln('<info>Jira Story Code</info>: ' . $story);
+        $output->writeln('<info>Story Name</info>: ' . $story_name);
+
```

Comment:
Seems that till here the code is a copy/paste from the gitBranchCommand, would be right to extend from the same base class and extract methods.